src/index.js

```diff
@@ -75,10 +75,12 @@ async function run(request, context) {
     // typically, this happens when the function isn't triggered by SQS but invoked "manually"
     return new Response('', { status: 204 });
   }
-
   await Promise.all(records.map(async ({ body }) => {
     try {
-      const payload = JSON.parse(body);
+      let payload = JSON.parse(body);
+      if (payload.Message) {
+        payload = JSON.parse(payload.Message);
+      }
```

Comment:
FIFO messages, have a `Message` property in the payload that actually contains **our** payload